thorlcr/activities/lookupjoin/thlookupjoinslave.cpp

```diff
@@ -730,17 +740,13 @@ class CInMemJoinBase : public CSlaveActivity, public CThorDataLink, public CAllO
         ~CRowProcessor()
         {
             blockQueue.stop();
-            loop
-            {
-                Owned<CSendItem> sendItem = blockQueue.dequeueNow();
-                if (NULL == sendItem)
-                    break;
-            }
+            clearQueue();
             wait();
         }
         void start()
         {
             stopped = false;
+            clearQueue();
```

Comment:
I don't think it makes a great deal either way..
Perhaps it would be more logical to clear on stop().
My thinking was that in the 1 run case clearQueue() will be called in ~CRowProcessor dtor anyway, so adding it to stop() would call it twice.
In >1 run case, it then _needs_ to be called.